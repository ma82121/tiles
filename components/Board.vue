<template>
  <div class="board">
    <TileView
      v-for="tile in tiles"
      :key="tile.index"
      :index="tile.index"
      :type="tile.type"
      :is-open="tile.isOpen"
      :has-debug-helper="useDebugHelper"
      :has-helper="useHelper"
      @clickTile="clickTile(tile)"
    />
  </div>
</template>

<script>
import TileView from '@/components/TileView'

export default {
  name: 'Board',
  components: {
    TileView
  },
  props: {
    tiles: {
      require: true,
      type: Array,
      default: () => []
    },
    useDebugHelper: {
      type: Boolean,
      default: false
    },
    useHelper: {
      type: Boolean,
      default: false
    }
  },
  methods: {
    clickTile(tile) {
      this.$emit('click', tile)
    }
  }
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/mixins/board.scss';

.board {
  @include board;
}
</style>
