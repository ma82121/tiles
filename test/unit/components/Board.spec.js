import { shallowMount } from '@vue/test-utils'
import Board from '@/components/Board'

describe('Board', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(Board)
    expect(wrapper.element).toMatchSnapshot()
  })
})
