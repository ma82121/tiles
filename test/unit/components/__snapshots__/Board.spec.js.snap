// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board snapshot 1`] = `
<div
  class="board"
/>
`;
